tests/model/component-import.cpp

```diff
@@ -0,0 +1,138 @@
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+/**
+ * @todo Need a way to have resources which can be used in tests, such as the
+ * source models used in these tests. But I guess not until we start validating
+ * models...
+ *
+ * While this is needed eventually, for now we are not instantiating the imports
+ * so don't need it just yet.
+ */
+
+TEST(ComponentImport, singleImport01) {
+    const std::string e =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+            "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+               "<import xlink:href=\"some-other-model.xml\" "
+                       "xmlns:xlink=\"http://www.w3.org/1999/xlink\">"
+                   "<component component_ref=\"a_component_in_that_model\" "
+                              "name=\"component_in_this_model\"/>"
+               "</import>"
+            "</model>";
+    libcellml::Model m;
+    libcellml::Import imp("some-other-model.xml");
+    m.addImport(imp);
+    libcellml::ComponentPtr importedComponent = std::make_shared<libcellml::Component>();
+    importedComponent->setName("component_in_this_model");
+    importedComponent->setSourceComponent("a_component_in_that_model");
+    imp.addComponent(importedComponent);
+
+    std::string a = m.serialise(libcellml::CELLML_FORMAT_XML);
+
+   EXPECT_EQ(e, a);
+}
+
+TEST(ComponentImport, singleImport02) {
+    const std::string e =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+            "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+               "<import xlink:href=\"some-other-model.xml\" "
+                       "xmlns:xlink=\"http://www.w3.org/1999/xlink\">"
+                   "<component component_ref=\"a_component_in_that_model\" "
+                              "name=\"component_in_this_model\"/>"
+               "</import>"
+            "</model>";
+    libcellml::Model m;
+    libcellml::Import imp("some-other-model.xml");
+
+    libcellml::ComponentPtr importedComponent = std::make_shared<libcellml::Component>();
+    importedComponent->setName("component_in_this_model");
+    importedComponent->setSourceComponent(imp, "a_component_in_that_model");
+    m.addComponent(importedComponent);
+
+    std::string a = m.serialise(libcellml::CELLML_FORMAT_XML);
+
+   EXPECT_EQ(e, a);
+}
+
+TEST(ComponentImport, multipleImport02) {
```

Comment:
Shouldn't this read:

```
TEST(ComponentImport, multipleImport01) {
```

since in this test, you explicitly add the import to the model (unlike in ```singleImport02```)?